From 99c28b5be02c533a3b14ff2f5ffa64b309900a66 Mon Sep 17 00:00:00 2001
From: David Wu <david.wu@rock-chips.com>
Date: Fri, 4 May 2018 17:30:14 +0800
Subject: [PATCH] arm64: dts: rockchip: Set pwm pin default pull setting when
 used for pwm regulator
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

As a second global reset, the GRF is not reset, the iomux and
pull of PWM pin is still keeping, but PWM controller is reset,
PWM pin goes into input mode. However, the pull is still none
changed in kernel, which can cause voltage problems, so should
always keep the PWM pin pull down mode, with 0~50 Î¼A power
increase.

Change-Id: Ibb1cbb5f5371d7838783264e23d5160c0757aaa6
Signed-off-by: David Wu <david.wu@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/rk3399-evb-ind.dtsi       |  2 ++
 arch/arm64/boot/dts/rockchip/rk3399-evb-rev3.dtsi      |  2 ++
 arch/arm64/boot/dts/rockchip/rk3399-evb.dts            |  2 ++
 .../arm64/boot/dts/rockchip/rk3399-firefly-android.dts |  2 ++
 arch/arm64/boot/dts/rockchip/rk3399-firefly.dts        |  2 ++
 .../arm64/boot/dts/rockchip/rk3399-mid-818-android.dts |  2 ++
 arch/arm64/boot/dts/rockchip/rk3399-rock960-ab.dts     |  2 ++
 arch/arm64/boot/dts/rockchip/rk3399-sapphire.dtsi      |  2 ++
 arch/arm64/boot/dts/rockchip/rk3399-tve1030g.dtsi      |  2 ++
 arch/arm64/boot/dts/rockchip/rk3399-tve1205g.dts       |  2 ++
 arch/arm64/boot/dts/rockchip/rk3399.dtsi               | 10 ++++++++++
 11 files changed, 30 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-evb-ind.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-evb-ind.dtsi
index 4813fc4d186b..227cc17a056e 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-evb-ind.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-evb-ind.dtsi
@@ -929,6 +929,8 @@
 
 &pwm2 {
 	status = "okay";
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm2_pin_pull_down>;
 };
 
 &pwm3 {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-evb-rev3.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-evb-rev3.dtsi
index 4c5f8a68d033..e34379ded3a5 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-evb-rev3.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-evb-rev3.dtsi
@@ -348,4 +348,6 @@
 
 &pwm2 {
 	status = "okay";
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm2_pin_pull_down>;
 };
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-evb.dts b/arch/arm64/boot/dts/rockchip/rk3399-evb.dts
index 959ddc3c7df5..7d19d5db4a36 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-evb.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-evb.dts
@@ -144,6 +144,8 @@
 
 &pwm3 {
 	status = "okay";
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm3a_pin_pull_down>;
 };
 
 &sdhci {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-firefly-android.dts b/arch/arm64/boot/dts/rockchip/rk3399-firefly-android.dts
index a26242da8b69..3d96d655d720 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-firefly-android.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-firefly-android.dts
@@ -931,6 +931,8 @@
 
 &pwm2 {
 	status = "okay";
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm2_pin_pull_down>;
 };
 
 &rkvdec {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-firefly.dts b/arch/arm64/boot/dts/rockchip/rk3399-firefly.dts
index 38336ab57cc4..0598b02f40d2 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-firefly.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-firefly.dts
@@ -639,6 +639,8 @@
 
 &pwm2 {
 	status = "okay";
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm2_pin_pull_down>;
 };
 
 &saradc {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-mid-818-android.dts b/arch/arm64/boot/dts/rockchip/rk3399-mid-818-android.dts
index 139b1b31fa81..bdeeb9739ba7 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-mid-818-android.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-mid-818-android.dts
@@ -919,6 +919,8 @@
 
 &pwm2 {
 	status = "okay";
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm2_pin_pull_down>;
 };
 
 &saradc {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-rock960-ab.dts b/arch/arm64/boot/dts/rockchip/rk3399-rock960-ab.dts
index f2ad4d08c9cd..a5ea9efd9872 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-rock960-ab.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-rock960-ab.dts
@@ -730,6 +730,8 @@
 
 &pwm2 {
 	status = "okay";
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm2_pin_pull_down>;
 };
 
 &pwm3 {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-sapphire.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-sapphire.dtsi
index e19a16a425ce..b4b2705f81bf 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-sapphire.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-sapphire.dtsi
@@ -508,6 +508,8 @@
 
 &pwm2 {
 	status = "okay";
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm2_pin_pull_down>;
 };
 
 &saradc {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-tve1030g.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-tve1030g.dtsi
index 7005f22f7edf..665c40d886b6 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-tve1030g.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-tve1030g.dtsi
@@ -802,6 +802,8 @@
 
 &pwm2 {
 	status = "okay";
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm2_pin_pull_down>;v
 };
 
 &rk_key {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-tve1205g.dts b/arch/arm64/boot/dts/rockchip/rk3399-tve1205g.dts
index 9239bc4f2aac..3a4d50da6f8a 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-tve1205g.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-tve1205g.dts
@@ -891,6 +891,8 @@
 
 &pwm2 {
 	status = "okay";
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm2_pin_pull_down>;
 };
 
 &route_edp {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399.dtsi b/arch/arm64/boot/dts/rockchip/rk3399.dtsi
index 19ef8ec0f841..a3cb1b21b0ac 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399.dtsi
@@ -2990,6 +2990,11 @@
 				rockchip,pins =
 					<0 RK_PA6 RK_FUNC_1 &pcfg_pull_none>;
 			};
+
+			pwm3a_pin_pull_down: pwm3a-pin-pull-down {
+				rockchip,pins =
+					<0 RK_PA6 RK_FUNC_1 &pcfg_pull_down>;
+			};
 		};
 
 		pwm3b {
@@ -2997,6 +3002,11 @@
 				rockchip,pins =
 					<1 RK_PB6 RK_FUNC_1 &pcfg_pull_none>;
 			};
+
+			pwm3b_pin_pull_down: pwm3b-pin-pull-down {
+				rockchip,pins =
+					<1 RK_PB6 RK_FUNC_1 &pcfg_pull_down>;
+			};
 		};
 
 		hdmi {
-- 
2.35.3

